pkg/skaffold/deploy/kubectl/cli.go

```diff
@@ -73,6 +84,67 @@ func (c *CLI) Apply(ctx context.Context, out io.Writer, manifests ManifestList)
 	return nil
 }
 
+type getResult struct {
+	Items []struct {
+		Metadata struct {
+			Name              string `json:"name"`
+			DeletionTimestamp string `json:"deletionTimestamp"`
+		} `json:"metadata"`
+	} `json:"items"`
+}
+
+// WaitForDeletions waits for resource marked for deletion to complete their deletion.
+func (c *CLI) WaitForDeletions(ctx context.Context, out io.Writer, manifests ManifestList) error {
+	previousList := ""
+	previousCount := 0
+
+	for try := 0; try < WaitDeletion.MaxRetry; try++ {
```

Comment:
Would it make sense to instead use a child context with a timeout?  (I'm really just concerned about infinite loops waiting for an object that will never die.)